### ISSUE TITLE
Add rule: no-promise-in-callback

### ISSUE THREAD

**Issue by pertrai1** [pertrai1/eslint-plugin-llm-core#10]
## Problem

LLMs mix Promises and callbacks incorrectly, returning Promises inside callback APIs.

## Rule

Flag functions that return Promises inside callback-style APIs.

### ❌ Bad
```ts
fs.readFile(path, (err, data) => {
  return processAsync(data); // Promise ignored!
});
```

### ✅ Good
```ts
fs.readFile(path, (err, data) => {
  processAsync(data).catch(handleError);
});
```

## Why it matters

- Mixing paradigms leads to unhandled rejections
- Callback APIs ignore returned Promises
- Commit to one style
